daffodil-cli/src/main/scala/org/apache/daffodil/Main.scala

```diff
@@ -880,7 +880,7 @@ object Main extends Logging {
     }
   }
 
-  def run(arguments: Array[String]): Int = {
+  def run(arguments: Array[String]): ExitCode.Value = {
```

Comment:
PropertyGenerator probably isn't a big deal. That won't have access to the Main.scala exit code unless we want to move the exit codes to lib. But many of these codes are very specific to the CLI, so I'm not sure that makes sense. The property generator also isn't ever run by users, it's just generates code as part of the buil.

The interactive debugger is also an interesting case. Although the interactive debugger is trigger from the CLI, it doesn't necessarily have to be, so a sys.exit() might even be wrong here. I wonder if instead it should throw some kind of debugger or interrupt exception that the CLI can catch and exit with an appropriate error code. And no CLI uses could handle it differently?

Reply:
Also, I would suggest that if the debugger exits via the "quit" command, that probably shouldn't be an error since that's correct behavior. So whatever happens when the debugger runs "quit", for the CLI it should probably result in a Success exit code.

Reply:
I would not change PropGen. The debugger can have a defined ExitCode or use generic ExitCode.Failure